### PR TITLE
1620 deletion of child ontologies

### LINKED ISSUE #1620: Deletion of child ontologies
We have no defined behaviour of the deletion of child ontologies.

Scenario:
When I upload a distributed ontology D with children C1, C2, ..., Cn
And I upload a new version of D1 with children C1, C2, ... C(i-1), C(i+1), ..., Cn
Then Ci should be inactive in the newer version, but active in an older version.

Currently, this makes our application fail (see #1593).

### PR DESCRIPTION
**Pull request by eugenk**
This shall fix #1620.

--

~~This branch is based on `1555-patch_url_for` of #1745 and **not based on staging**. It may be easier to review that one first.~~

### CODE REVIEW

File: app/models/ontology/associations_and_attributes.rb [hunk 1]
```diff
@@ -2,6 +2,11 @@ module Ontology::AssociationsAndAttributes
   extend ActiveSupport::Concern
 
   included do
+    acts_as_tree
+    # Add the scope "present: true" to the children association by acts_as_tree.
+    Ontology.reflect_on_association(:children).
```
Comment (eugenk) [inline app/models/ontology/associations_and_attributes.rb]: This is done (rebase, commit amend).

Comment (eugenk) [inline app/models/ontology/associations_and_attributes.rb]: This is done (rebase, commit amend).

File: lib/hets/dg/node_evaluation_helper.rb [hunk 2]
```diff
@@ -149,6 +149,16 @@ def update_ontologies_per_logic_count!(ontologies)
           Logic.reset_counters(logic_id, :ontologies)
         end
       end
+
+      def set_all_children_inactive_except(present_ontologies)
```
Comment (eugenk) [inline lib/hets/dg/node_evaluation_helper.rb]: No, thanks.

**Comment by ebolloff** [comment]
When I update an ontology and update it to the initial version after that I get the following error

```
PG::UniqueViolation: FEHLER:  doppelter Schlüsselwert verletzt Unique-Constraint „index_loc_ids_on_locid“
DETAIL:  Schlüssel „(locid)=(/default/test2//sp1)“ existiert bereits.
: INSERT INTO "loc_ids" ("created_at", "locid", "specific_id", "specific_type", "updated_at") VALUES ($1, $2, $3, $4, $5) RETURNING "id"
```

**Comment by ebolloff** [comment]
The error still occurs.

**Comment by eugenk** [comment]
@ebolloff, this should work now.

**Comment by ebolloff** [comment]
The tests hang up on my machine, but they work on @phyrog and @eugenk machines so :+1:

**Pull request merged**